tmt/steps/report/reportportal.py

```diff
@@ -590,15 +607,19 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
                             response = self.rp_api_post(
                                 session=session,
                                 path="log/entry",
-                                json={"message": _filter_invalid_chars(log),
+                                json={"message": _filter_invalid_chars(
+                                    _filter_log_per_size(log,
+                                                         int(self.data.log_size_limit))),
```

Comment:
It was one filter at first, but now we have two, which means there can easily be seven next month :) How about detaching them from the code that needs to call them? With some simple "registry":

```python
def _filter_invalid_chars(data: str, step_data: 'ReportReportPortalData') -> str:
    return re.sub(
        '[^\u0020-\uD7FF\u0009\u000A\u000D\uE000-\uFFFD\U00010000-\U0010FFFF]+',
        '',
        data)


def _filter_log_per_size(data: str, step_data: 'ReportReportPortalData') -> str:
    size = len(log)
    if size > step_data.log_size_limit:
        return (f"Log file was not uploaded to ReportPortal because its size {size} bytes"
                f"exceeds {step_data.log_size_limit} bytes.")
    return log


_LOG_FILTERS = [
  _filter_invalid_chars,
  _filter_log_per_size
]


def _filter_log(self, log: str) -> str:
  for log_filter in _LOG_FILTERS:
    log = log_filter(log, self.data)
  return log

message = self._filter_log(log)
```

Reply:
Updated.